src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -76,9 +75,7 @@ export default function IndexPatternFactory(Private, timefilter, Notifier, confi
       .id(self.id);
 
       // listen for config changes and update field list
-      $rootScope.$on('change:config', function () {
-        initFields();
-      });
+      config.on('*', initFields);
```

Comment:
I think this is going to cause issues because it's passing the settings as the first argument to initFields